components/proxy/src/main/java/com/hotels/styx/admin/handlers/ProxyStatusHandler.java

```diff
@@ -0,0 +1,52 @@
+/*
+  Copyright (C) 2013-2018 Expedia Inc.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+package com.hotels.styx.admin.handlers;
+
+import com.hotels.styx.api.HttpRequest;
+import com.hotels.styx.api.HttpResponse;
+import com.hotels.styx.api.http.handlers.BaseHttpHandler;
+import com.hotels.styx.events.Event;
+import com.hotels.styx.events.EventNexus;
+import org.slf4j.Logger;
+
+import static com.hotels.styx.api.HttpResponse.Builder.response;
+import static org.slf4j.LoggerFactory.getLogger;
+
+/**
+ * Handler that returns whether the proxy server has started.
+ */
+public class ProxyStatusHandler extends BaseHttpHandler {
+    private static final Logger LOGGER = getLogger(ProxyStatusHandler.class);
+
+    private volatile boolean proxyServerStarted;
+
+    public ProxyStatusHandler(EventNexus eventNexus) {
+        eventNexus.events("server.started.proxy")
+                .map(Event::value)
+                .cast(Boolean.class)
+                .subscribe(started -> {
+                    proxyServerStarted = started;
+                    LOGGER.info("proxyServerStarted={}", started);
+                });
+    }
```

Comment:
Another issue with "server.started.proxy". Styx is capable of starting HTTP and HTTPS ports separately. There could be a possibility, or at least we need to consider a possibility, of them starting somewhat off-sync. Therefore we perhaps need to think about exposing them separately via this admin interface.

Reply:
The code is only considering the proxy started when *everything* is ready. I don't think there is a use-case for an endpoint saying when Styx is partially ready.

Reply:
Possibly not.